### PR TITLE
fix: missing metadata dependency

### LINKED ISSUE #2825: `System.Reflection.Metadata` dependency on v4.0
[Originally reported on Discord](https://discord.com/channels/621778831602221064/1172378051074609242)

> With uwp it seems like it requires another extra library that wasn't installed by default

```
System.IO.FileNotFoundException: Could not load file or assembly 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
File name: 'System.Reflection.Metadata, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at System.Diagnostics.Internal.PortablePdbReader..ctor()
   at System.Diagnostics.EnhancedStackTrace.GetFrames(StackTrace stackTrace)
   at Sentry.Internal.DebugStackTrace.CreateFrames(StackTrace stackTrace, Boolean isCurrentStackTrace)+MoveNext()
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ReverseIterator`1.MoveNext()
   at Sentry.Internal.DebugStackTrace.Create(SentryOptions options, StackTrace stackTrace, Boolean isCurrentStackTrace)
   at Sentry.Extensibility.SentryStackTraceFactory.Create(Exception exception)
   at Sentry.Internal.MainExceptionProcessor.BuildSentryException(Exception exception, Int32 id, Nullable`1 parentId, String source)
   at Sentry.Internal.MainExceptionProcessor.WalkExceptions(Exception exception, Counter counter, Nullable`1 parentId, String source)+MoveNext()
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ReverseIterator`1.MoveNext()
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Sentry.Internal.MainExceptionProcessor.CreateSentryExceptions(Exception exception)
   at Sentry.Internal.MainExceptionProcessor.Process(Exception exception, SentryEvent sentryEvent)
```

### PR DESCRIPTION
**Pull request by vaind**
Fixes #2825

#skip-changelog - only part of the alpha release

### CODE REVIEW

File: src/Sentry/Internal/Enricher.cs [hunk 1]
```diff
@@ -33,10 +33,14 @@ public void Apply(IEventLike eventLike)
         // Operating System
         if (!eventLike.Contexts.ContainsKey(OperatingSystem.Type))
         {
-            // RuntimeInformation.OSDescription is throwing on Mono 5.12
-            if (!PlatformAbstractions.Runtime.Current.IsMono())
-            {
-                eventLike.Contexts.OperatingSystem.RawDescription = RuntimeInformation.OSDescription;
+            // RuntimeInformation.OSDescription is throwing on Mono 5.12 and .net 4.6.2 on Linux/macOS
+            try {
+                if (!PlatformAbstractions.Runtime.Current.IsMono())
```
Comment (bitsandfoxes) [inline src/Sentry/Internal/Enricher.cs:38]: Is the check now throwing too?

**Review by bitsandfoxes — APPROVED** [review]

File: src/Sentry/Internal/Enricher.cs (hunk 1 above)
Comment (vaind) [reply src/Sentry/Internal/Enricher.cs:38]: the check could be outside of try-catch, it's the call below that throws on net4.6.2 on Linux and macOS.

File: src/Sentry/Internal/Enricher.cs [hunk 2]
```diff
@@ -33,10 +33,14 @@ public void Apply(IEventLike eventLike)
...
+                {
+                    eventLike.Contexts.OperatingSystem.RawDescription = RuntimeInformation.OSDescription;
+                }
+            } catch {
+                // ignored
```
Comment (vaind) [inline src/Sentry/Internal/Enricher.cs:43]:
```suggestion
            if (!PlatformAbstractions.Runtime.Current.IsMono())
            {
                try {
                    eventLike.Contexts.OperatingSystem.RawDescription = RuntimeInformation.OSDescription;
                } catch {
                    // ignored
                }
```

File: src/Sentry/Internal/Enricher.cs (hunk 1 above)
Comment (vaind) [reply src/Sentry/Internal/Enricher.cs]: anyway... let me check if it throws on netstandard too and I'll add an ifdef

**Pull request merged**